beacon_node/lighthouse_network/src/rpc/codec.rs

```diff
@@ -359,6 +360,11 @@ impl<E: EthSpec> Encoder<OutboundRequest<E>> for SSZSnappyOutboundCodec<E> {
 
         // Write compressed bytes to `dst`
         dst.extend_from_slice(writer.get_ref());
+        metrics::inc_counter_vec_by(
+            &metrics::TOTAL_RPC_REQUESTS_BYTES_SENT,
+            &[item.into()],
+            dst.len() as u64,
+        );
```

Comment:
Feels wrong to place a metric inside an `encode` function in a code struct, may be best to place this on the consumer of this function?